docs/csharp/tutorials/exploration/csharp-6.yml

```
@@ -0,0 +1,240 @@
+### YamlMime:Tutorial
+title: Explore C# 6 - C# interactive C# tutorial
+metadata:
+  title: Explore C# 6 - Try the new features in C# 6 interactively, using your browser
+  description: In this tutorial, you'll use your browser to explore C# 6 interactively. You'll explore the new idioms you can use with C# 6 that enable more concise and readable code.
+  audience: Developer
+  level: intermediate
+  ms.date: 12/11/2018
+  displayType: two-column
+  interactive: csharp
+items:
+- durationInMinutes: 1
+  content: |
+    This tutorial lets you explore C# 6 interactively, using your browser to write C# and see the results of compiling and running your code. It contains a series of lessons that modify earlier C# practices to use newer, more concise C# 6 features. The rest of this article provides an overview of each of these features, with a link to explore each feature. You can read about the features in the [What's new in C# 6](../whats-new/csharp-6.md) article in the "What's new" section.
+
+- title: Read only auto-properties enable readonly types
+  durationInMinutes: 2
+  content: |
+    Two enhancements to auto-property syntax make it easier to use auto-properties in more places: read-only auto-properties and auto-property initializers. Consider this small program:
+
+    [!code-csharp[Starter](../../../../samples/csharp/tutorials/explorations/csharp6-starter/Program.cs)]  
+
+        
+    Enter Focus mode, copy the preceding code into the C# interactive window. Then, click *Run* to see the error. The author of the `Person` class intended the strings for `FirstName` and `LastName` to be readonly. With C# 6, you can make that intent clear. Remove the `private set` from both properties to create a reaonly auto property. Click *Run* to see that the compiler spots the two locations where the `FirstName` and `LastName` properties are changed when they should not have been. You can change the `AllCaps` method to the following code to fix the compiler error:
```

also should auto-property be hyphened?

```suggestion
     Enter Focus mode, copy the preceding code into the C# interactive window. Then, click *Run* to see the error. The author of the `Person` class intended the strings for `FirstName` and `LastName` to be read-only. With C# 6, you can make that intent clear. Remove the `private set` from both properties to create a read-only auto property. Click *Run* to see that the compiler spots the two locations where the `FirstName` and `LastName` properties are changed when they should not have been. You can change the `AllCaps` method to the following code to fix the compiler error:
```